proto/monitor_service.proto

```diff
@@ -31,8 +31,28 @@ message HeapProfilingRequest {
 
 message HeapProfilingResponse {}
 
+message ListHeapProfilingRequest {}
+message ListHeapProfilingResponse {
+  string dir_manually = 1;
+  repeated string name_manually = 2;
+  string dir_auto = 3;
+  repeated string name_auto = 4;
+}
+
+// Download files
+message DownloadRequest {
+  // The file path
+  string path = 1;
+}
+
+message DownloadResponse {
+  bytes result = 1;
+}
+
 service MonitorService {
   rpc StackTrace(StackTraceRequest) returns (StackTraceResponse);
   rpc Profiling(ProfilingRequest) returns (ProfilingResponse);
   rpc HeapProfiling(HeapProfilingRequest) returns (HeapProfilingResponse);
+  rpc ListHeapProfiling(ListHeapProfilingRequest) returns (ListHeapProfilingResponse);
+  rpc Download(DownloadRequest) returns (DownloadResponse);
```

Comment:
Is it possible to directly return the dump result (`bytes`) in the response of profiling, instead of using a separate phase to download the file?

Reply:
Download is also used to download auto dumped files. This can unify the path of getting auto and manually dumped files